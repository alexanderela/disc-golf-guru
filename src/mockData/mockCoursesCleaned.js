const mockCoursesCleaned = [
  {
    id: '4712',
    name: 'Genesee Valley Park',
    isFavorite: false,
    holes: '18',
    city: 'Rochester',
    state: 'NY',
    country: 'United States',
    zip: '14620',
    address: '99 Elmwood Ave.',
    rating: '3.29',
    isPayToPlay: 'No',
    isPrivate: 'No',
  },
  {
    id: '7628',
    name: 'Lookup Park',
    isFavorite: false,
    holes: '9',
    city: 'Henrietta',
    state: 'NY',
    country: 'United States',
    zip: '14467',
    address: '3850 East Henrietta Rd.',
    rating: '3.13',
    isPayToPlay: 'No',
    isPrivate: 'No',
  },
  {
    id: '4937',
    name: 'Monroe Community College DGC',
    isFavorite: false,
    holes: '18',
    city: 'Henrietta',
    state: 'NY',
    country: 'United States',
    zip: '14623',
    address: '',
    rating: '2.94',
    isPayToPlay: 'No',
    isPrivate: 'No',
  },
  {
    id: '6795',
    name: 'Ellison Park',
    isFavorite: false,
    holes: '18',
    city: 'Rochester',
    state: 'NY',
    country: 'United States',
    zip: '14610',
    address: '',
    rating: '3.54',
    isPayToPlay: 'No',
    isPrivate: 'No',
  },
  {
    id: '6443',
    name: 'North Ponds DGC',
    isFavorite: false,
    holes: '9',
    city: 'Webster',
    state: 'NY',
    country: 'United States',
    zip: '14580',
    address: '750 Holt Rd.',
    rating: '1.75',
    isPayToPlay: 'No',
    isPrivate: 'No',
  },
];

export default mockCoursesCleaned;
