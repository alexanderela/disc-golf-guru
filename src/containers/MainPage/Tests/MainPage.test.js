import React from 'react';
import ReactDOM from 'react-dom';
import { Switch } from 'react-router-dom';
import { MainPage } from '../';
import { mapStateToProps, mapDispatchToProps } from '../';
import { shallow, mount } from 'enzyme';
import mockCoursesCleaned from '../../../mockData/mockCoursesCleaned.js';
jest.mock('../../../utilities/API');

describe('MainPage', () => {
	let mockFunc;
	let wrapper;
	let mockFetchGolfCourses;
	let mockToggleSearch;
	let mockToggleCourse;

	beforeEach(() => {
		mockFunc = jest.fn();
		mockFetchGolfCourses = jest.fn();
		mockToggleSearch = jest.fn();
		mockToggleCourse = jest.fn();
		wrapper = shallow(
									<MainPage 
										fetchGolfCourses={mockFetchGolfCourses} 
										golfCourses={mockCoursesCleaned}
										toggleSearchResults={mockToggleSearch}
										toggleCourseDetails={mockToggleCourse}/>);
	})

	it('should render like snapshot', () => {
		expect(wrapper).toMatchSnapshot()
	})
	
	describe('handleInputChange', () => {
		let mockEvent;

		beforeEach(() => {
			mockEvent = { target: { value: 'white plains'} }
		})

		it('should call handleInputChange when searchTerms are changed', () => {
			const spy = spyOn(wrapper.instance(), 'handleInputChange');
			wrapper.instance().forceUpdate();
			wrapper.find('.search-input').simulate('change', mockEvent)
			expect(spy).toHaveBeenCalled()
		})

		it('should update state', () => {
			wrapper.instance().handleInputChange(mockEvent);
			expect(wrapper.state('searchTerms')).toBe('white plains')
		})
	})

	describe('handleSubmit', () => {
		let mockEvent;

		beforeEach(() => {
			mockEvent = { preventDefault: jest.fn() }
		})
		
		it('should call handleSubmit upon submission of form', () => {
			const spy = spyOn(wrapper.instance(), 'handleSubmit');
			wrapper.instance().forceUpdate();
			wrapper.find('form').simulate('submit', mockEvent)
			expect(spy).toHaveBeenCalled();
		})

		it('should call getGolfCourses when handleSubmit is called', () => {
			wrapper.instance().getGolfCourses = jest.fn();
			wrapper.instance().handleSubmit(mockEvent);
			expect(wrapper.instance().getGolfCourses).toHaveBeenCalled();
		})

		it('should update state', () => {
			wrapper.setState({ searchTerms: 'white plains' })
			wrapper.instance().handleSubmit(mockEvent);
			expect(wrapper.state('searchTerms')).toBe('');
		})
	})

	describe('getGolfCourses', () => {
		let mockZip;

		beforeEach(() => {
			mockZip = 14526;
		})

		it('should call fetchGolfCourses when getGolfCourses is called', async () => {
			await wrapper.instance().getGolfCourses(mockZip);
			expect(mockFetchGolfCourses).toHaveBeenCalled();
		})

		it('should call toggleSearchResults when getGolfCourses is called', async () => {
			await wrapper.instance().getGolfCourses(mockZip);
			expect(mockToggleSearch).toHaveBeenCalled();
		})
	})

	describe('mapStateToProps', () => {
		it('should create the correct props object', () => {
			const expected = {
				golfCourses: [{ id: 1234, name: 'mockCourse1'}, { id: 4321, name: 'mockCourse2'}]
			}
			const result = mapStateToProps(expected);
			expect(result).toEqual(expected);
		})
	})

	describe('mapDispatchToProps', () => {
		let dispatch;

		beforeEach(() => {
			dispatch = jest.fn();
		})

		it('should map a key of fetchGolfCourses', () => {
			const dispatchedProps = mapDispatchToProps(dispatch);
			expect(dispatchedProps.fetchGolfCourses).toBeDefined()
		})

		it('should map a key of toggleSearchResults', () => {
			const dispatchedProps = mapDispatchToProps(dispatch);
			expect(dispatchedProps.toggleSearchResults).toBeDefined()
		})

		it('should map a key of toggleCourseDetails', () => {
			const dispatchedProps = mapDispatchToProps(dispatch);
			expect(dispatchedProps.toggleCourseDetails).toBeDefined()
		})

		it('should have fetchGolfCourses call dispatch', () => {
			const dispatchedProps = mapDispatchToProps(dispatch);
			dispatchedProps.fetchGolfCourses(mockCoursesCleaned);
			expect(dispatch).toHaveBeenCalled();
		})

		it('toggleSearchResults should call dispatch', () => {
      const dispatchProps = mapDispatchToProps(dispatch);
      dispatchProps.toggleSearchResults();
      expect(dispatch).toHaveBeenCalled();
    });

  	it('toggleCourseDetails should call dispatch', () => {
      const dispatchProps = mapDispatchToProps(dispatch);
      dispatchProps.toggleCourseDetails();
      expect(dispatch).toHaveBeenCalled();
    });
	})

})