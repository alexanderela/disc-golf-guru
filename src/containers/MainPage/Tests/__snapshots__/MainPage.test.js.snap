// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage should render like snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainPage
    fetchGolfCourses={[MockFunction]}
    golfCourses={
      Array [
        Object {
          "address": "99 Elmwood Ave.",
          "city": "Rochester",
          "country": "United States",
          "holes": "18",
          "id": "4712",
          "isFavorite": false,
          "isPayToPlay": "No",
          "isPrivate": "No",
          "name": "Genesee Valley Park",
          "rating": "3.29",
          "state": "NY",
          "zip": "14620",
        },
        Object {
          "address": "3850 East Henrietta Rd.",
          "city": "Henrietta",
          "country": "United States",
          "holes": "9",
          "id": "7628",
          "isFavorite": false,
          "isPayToPlay": "No",
          "isPrivate": "No",
          "name": "Lookup Park",
          "rating": "3.13",
          "state": "NY",
          "zip": "14467",
        },
        Object {
          "address": "",
          "city": "Henrietta",
          "country": "United States",
          "holes": "18",
          "id": "4937",
          "isFavorite": false,
          "isPayToPlay": "No",
          "isPrivate": "No",
          "name": "Monroe Community College DGC",
          "rating": "2.94",
          "state": "NY",
          "zip": "14623",
        },
        Object {
          "address": "",
          "city": "Rochester",
          "country": "United States",
          "holes": "18",
          "id": "6795",
          "isFavorite": false,
          "isPayToPlay": "No",
          "isPrivate": "No",
          "name": "Ellison Park",
          "rating": "3.54",
          "state": "NY",
          "zip": "14610",
        },
        Object {
          "address": "750 Holt Rd.",
          "city": "Webster",
          "country": "United States",
          "holes": "9",
          "id": "6443",
          "isFavorite": false,
          "isPayToPlay": "No",
          "isPrivate": "No",
          "name": "North Ponds DGC",
          "rating": "1.75",
          "state": "NY",
          "zip": "14580",
        },
      ]
    }
    toggleCourseDetails={[MockFunction]}
    toggleSearchResults={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="page-name"
        />,
        <input
          className="search-input"
          onChange={[Function]}
          placeholder="Search for a zip code or city"
          type="search"
          value=""
        />,
        <button
          className="submit-btn"
          type="submit"
        >
          Submit
        </button>,
        <Route
          exact={true}
          path="/findcourses"
          render={[Function]}
        />,
        <Route
          exact={true}
          path="/findcourses/searchresults"
          render={[Function]}
        />,
        <Route
          exact={true}
          path="/findcourses/searchresults/courseinfo/:id"
          render={[Function]}
        />,
      ],
      "className": "MainPage",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "page-name",
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "search-input",
          "onChange": [Function],
          "placeholder": "Search for a zip code or city",
          "type": "search",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Submit",
          "className": "submit-btn",
          "type": "submit",
        },
        "ref": null,
        "rendered": "Submit",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/findcourses",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/findcourses/searchresults",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/findcourses/searchresults/courseinfo/:id",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="page-name"
          />,
          <input
            className="search-input"
            onChange={[Function]}
            placeholder="Search for a zip code or city"
            type="search"
            value=""
          />,
          <button
            className="submit-btn"
            type="submit"
          >
            Submit
          </button>,
          <Route
            exact={true}
            path="/findcourses"
            render={[Function]}
          />,
          <Route
            exact={true}
            path="/findcourses/searchresults"
            render={[Function]}
          />,
          <Route
            exact={true}
            path="/findcourses/searchresults/courseinfo/:id"
            render={[Function]}
          />,
        ],
        "className": "MainPage",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "page-name",
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "search-input",
            "onChange": [Function],
            "placeholder": "Search for a zip code or city",
            "type": "search",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Submit",
            "className": "submit-btn",
            "type": "submit",
          },
          "ref": null,
          "rendered": "Submit",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/findcourses",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/findcourses/searchresults",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/findcourses/searchresults/courseinfo/:id",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
